google_sso/README.md

```diff
@@ -0,0 +1,307 @@
+allows enabling SSO experiences for certain Perforce products using either OpenID Connect
+and SAML 2.0 authentication protocols.
+
+More information about HAS can be found here:
+https://www.perforce.com/downloads/helix-authentication-service
+
+This guide provides step-by-step instructions on how to install a single instance of the
+Helix Core software on the Google Cloud Platform and how to configure HAS using OpenID Connect,
+enabling users to sign into Helix Core using Google Identity
+(such as a Gmail or a Google Workspaces account).
+
+Please note that this guide is taking a number of shortcuts to keep it simple. The last section
+of this guide will focus on some important considerations regarding this setup.
+
+## Prerequisites
+
+In order to complete the steps in this guide, you would need a Google Cloud account
+with billing enabled. The Google Cloud Platform offers a free trial that should be sufficient to
+cover all of the steps in this guide.
+
+## Setting up Helix Core with HAS and Google Identity
+
+### Creating a GCE VM to host Helix Core and HAS
+
+We will be using a Google Compute Engine virtual machine to install both Helix Core and HAS.
+
+Create a new VM by navigating to the Google Cloud Console, selecting Compute Engine and clicking "Create Instance".
+
+Select an E2-medium instance with the Ubuntu 18.04 LTS image and a 100GB HDD.
+
+_Note that the machine specs are mostly artbitrary - HAS doesn't require powerful VMs
+but you might consider more powerful specs if you were to actually run Helix Core in production._
+
+![Create Instance](create_instance.png)
+
+Alternatively, you can create the instance using the gcloud command:
+
+```
+gcloud compute instances create perforce --zone=us-central1-a --machine-type=e2-medium --image=ubuntu-1804-bionic-v20210129 --image-project=ubuntu-os-cloud --boot-disk-size=100GB --boot-disk-type=pd-standard
+```
+
+Please note that for the purposes of keeping this simple, the VM has been configuring using
+an ephemeral external IP. In an actual production environment, you'll probably want to assign
+a permanent external IP and use it in conjuction with a custom domain name.
+
+Take a note of the VM's external IP address, you will need it in the subsequent steps.
+We will refer to is as VM_EXTERNAL_IP in the rest of the guide.
+
+### Configuring OAuth
+
+In order to use HAS with Google Identity, you will need to configure an OAuth 2.0 client.
+In the Google Cloud Console, navigate to API & Services and select Credentials.
+Click on Create Credentials, select "OAuth client ID" and pick "Web application"
+as the Application type:
+
+![Create Credentials](create_credentials.png)
+
+You can name the credentials "HAS Client" (the name does not matter).
+
+Note that you will need to provide the authorized Authorized redirect URIs.
+Since we are not using a domain name but an IP address, we would need to use a special domain,
```

Comment:
Add a recommendation that users should use their own domainname for a production environment.

Reply:
Added the recommendation